stdlib/Test/src/precompile.jl

```diff
@@ -0,0 +1,31 @@
+precompile(Tuple{typeof(Test.do_test), Test.ExecutionResult, Any})
+precompile(Tuple{typeof(Test.testset_beginend_call), Tuple{String, Expr}, Expr, LineNumberNode})
+precompile(Tuple{Type{Test.DefaultTestSet}, String})
+precompile(Tuple{Type{Test.DefaultTestSet}, AbstractString})
+precompile(Tuple{Core.kwftype(Type{Test.DefaultTestSet}), Any, Type{Test.DefaultTestSet}, AbstractString})
+precompile(Tuple{typeof(Test.finish), Test.DefaultTestSet})
+precompile(Tuple{typeof(Test.eval_test), Expr, Expr, LineNumberNode, Bool})
+precompile(Tuple{typeof(Test._inferred), Expr, Module})
+precompile(Tuple{typeof(Test.push_testset), Test.DefaultTestSet})
+precompile(Tuple{typeof(Test.get_alignment), Test.DefaultTestSet, Int})
+precompile(Tuple{typeof(Test.get_test_result), Any, Any})
+precompile(Tuple{typeof(Test.do_test_throws), Test.ExecutionResult, Any, Any})
+precompile(Tuple{typeof(Test.print_counts), Test.DefaultTestSet, Int, Int, Int, Int, Int, Int, Int})
+precompile(Tuple{typeof(Test._check_testset), Type, Expr})
+precompile(Tuple{typeof(Test.test_expr!), Any, Any})
+precompile(Tuple{typeof(Test.test_expr!), Any, Any, Vararg{Any, 100}})
+precompile(Tuple{typeof(Test.pop_testset)})
+precompile(Tuple{typeof(Test.match_logs), Function, Tuple{Symbol, Regex}})
+precompile(Tuple{typeof(Test.match_logs), Function, Tuple{String, Regex}})
+precompile(Tuple{typeof(Base.CoreLogging.shouldlog), Test.TestLogger, Base.CoreLogging.LogLevel, Module, Symbol, Symbol})
+precompile(Tuple{typeof(Base.CoreLogging.handle_message), Test.TestLogger, Base.CoreLogging.LogLevel, String, Module, Symbol, Symbol, String, Int})
+precompile(Tuple{typeof(Test.detect_ambiguities), Any})
+precompile(Tuple{typeof(Test.collect_test_logs), Function})
+precompile(Tuple{typeof(Test.do_broken_test), Test.ExecutionResult, Any})
+precompile(Tuple{typeof(Test.record), Test.DefaultTestSet, Union{Test.Error, Test.Fail}})
+precompile(Tuple{typeof(Test.filter_errors), Test.DefaultTestSet})
+# TODO: Check which of the explicit precompile statments above are still needed
+# with the workload below active.
+@testset "example" begin
```

Comment:
This prints to stderr which is not so pretty

Reply:
Can stderr be redirected or would that ruin the precompilation effectiveness?

Reply:
They all print from the same worker so maybe there is some test running a bunch of processes with `--compiled-modules=no` that loads Test?

Reply:
It is also printed twice when compiling Julia:

```
    JULIA stdlib/DelimitedFiles.release.image
    JULIA stdlib/LibSSH2_jll.release.image
    JULIA stdlib/SharedArrays.release.image
    JULIA stdlib/Profile.release.image
Test Summary: | Pass  Total  Time
example       |    1      1  0.1s
    JULIA stdlib/Test.release.image
    JULIA stdlib/SparseArrays.release.image
Test Summary: | Pass  Total  Time
example       |    1      1  0.1s
```

Seems it should only be once (when it is compiled)?

Reply:
Oh yeah, one with bounds and one without. So two total

Reply:
(it's the loading tests that prints it)

Reply:
Perhaps all these precompiles should be guarded to happen during precompilation only like SnoopCompile ones are?